keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/diff/DualNumber.java

```diff
@@ -395,22 +397,25 @@ public DualNumber reshape(int[] proposedShape) {
     }
 
     public DualNumber slice(int dimension, int index) {
-        PartialDerivatives slicedPartialDerivatives = this.partialDerivatives.slice(dimension, index);
-        return new DualNumber(value.slice(dimension, index), slicedPartialDerivatives);
+        DoubleTensor newValue = value.slice(dimension, index);
+        boolean needReshape = newValue.getRank() == value.getRank();
```

Comment:
This would be the case for 2,1,0 but we don't actually use 1 and 0. I also like it's more general in that you need to reshape *because* there wasn't a rank change, which if slice was correct there would be.

Reply:
Cool let's keep it as it is